app/src/main/kotlin/com/wire/android/navigation/NavigationManager.kt

```diff
@@ -21,18 +21,30 @@
 package com.wire.android.navigation
 
 import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.take
 
 class NavigationManager {
 
-    var navigateState = MutableSharedFlow<NavigationCommand?>()
-    var navigateBack = MutableSharedFlow<Map<String, Any>>()
+    private val _navigateState = MutableSharedFlow<NavigationCommand?>()
+    private val _navigateBack = MutableSharedFlow<Map<String, Any>>()
+    var navigateState: SharedFlow<NavigationCommand?> = _navigateState
+    var navigateBack: SharedFlow<Map<String, Any>> = _navigateBack
 
     suspend fun navigate(command: NavigationCommand) {
-        navigateState.emit(command)
+        if (_navigateState.subscriptionCount.value > 0) {
+            _navigateState.emit(command)
+        } else {
+            _navigateState.subscriptionCount
+                .filter { it > 0 }
+                .take(1)
+                .collect { _navigateState.emit(command) }
+        }
```

Comment:
Could we extract to a separate method or at least add some documentation about what is this code doing?